### PR TITLE
Add Transport.RoundTrip()

### PR DESCRIPTION
**Pull request by sn00pyd0g3**
Add the function #504

### CODE REVIEW

**Comment by smowton** [comment]
You don't need to open a new PR to do a rebase -
see https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase
and https://www.atlassian.com/git/tutorials/syncing/git-push using `--force-with-lease` to replace the PR branch with the rebased version.

**Comment by sn00pyd0g3** [comment]
> You don't need to open a new PR to do a rebase -
> see https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase
> and https://www.atlassian.com/git/tutorials/syncing/git-push using `--force-with-lease` to replace the PR branch with the rebased version.

ok,I will do it next time. the old PR was closed, use this one

File: ql/src/semmle/go/frameworks/stdlib/NetHttp.qll [hunk 1]
```diff
@@ -216,6 +216,35 @@ module NetHttp {
...
 
+  /** A call to the `Transport.RoundTrip` function in the `net/http` package. */
+  private class TransportRoundTrip extends HTTP::ClientRequest::Range, DataFlow::MethodCallNode {
+    TransportRoundTrip() {
+      this.getTarget().(Method).hasQualifiedName("net/http", "Transport", "RoundTrip")
+    }
+
+    override DataFlow::Node getUrl() {
+      // A URL passed to `NewRequest`, whose result is passed to this `RoundTrip` call
+      exists(DataFlow::CallNode call | call.getTarget().hasQualifiedName("net/http", "NewRequest") |
```
Comment (sauyon) [inline ql/src/semmle/go/frameworks/stdlib/NetHttp.qll:227]: I imagine that `Transport`s aren't made with `NewRequest`?

Comment (sn00pyd0g3) [reply ql/src/semmle/go/frameworks/stdlib/NetHttp.qll:227]: maybe it's not designed for this... but it's work😹 anyway, it's a SSRF sink

Comment (sauyon) [reply ql/src/semmle/go/frameworks/stdlib/NetHttp.qll:227]: No, this was just me not thinking...

**Review by sauyon — COMMENTED** [review]
LGTM, but the code isn't autoformatted; you should run `make autoformat`.

**Comment by sauyon** [comment]
Could you add a change note? See the format in the `change-notes` folder.

**Comment by sn00pyd0g3** [comment]
> Could you add a change note? See the format in the `change-notes` folder.

done

File: change-notes/2021-03-16-nethttp-updated.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+lgtm,codescanning
+* Added TransportRoundTrip in NetHttp.qll, it can performs an HTTP request to a URL.
```
Comment (sauyon) [inline change-notes/2021-03-16-nethttp-updated.md:2]:
```suggestion
* Added support for the `Transport.RoundTrip` method in `net/http`.
```

**Review by sauyon — COMMENTED** [review]
This should be the last thing, sorry.

**Review by sauyon — APPROVED** [review]
👍 thanks

**Pull request merged**